protected/models/Project.php

```diff
@@ -109,19 +111,16 @@ public static function getListProjects(){
     }
     
     function check_duplicate(){
-        
-        
         $db_url= Project::model()->findBySql("select name from project where url='$this->url'");
         
-        if($db_url !=null){
-        $this->addError('url','Duplicate URL');}
+        if ($db_url !=null) {
+            $this->addError('url','Duplicate URL');
+        }
         
         $db_name= Project::model()->findBySql("select url from project where name='$this->name'");
        
-        if($db_name !=null){
-        $this->addError('name','Duplicate Project Name');}
-        
-             
+        if ($db_name !=null) {
+            $this->addError('name','Duplicate Project Name');
+        }
```

Comment:
**question:** This function does not handle the case insensitive case (e.g., example.com and Example.com would be treated as different), if we were going to improve it, should we consider adding case-insensitive checks, such as using LOWER() in the query as below:
```
$db_url=Project::model()->findBySql("SELECT * FROM project WHERE LOWER(url) = LOWER('$this->url');
$db_name=Project::model()->findBySql("SELECT * FROM project WHERE LOWER(name) = LOWER('$this->name');
```

**suggestion:** Actually, I think we could use Yii's built-in validator called [`unique`](https://www.yiiframework.com/wiki/56/reference-model-rules-validation#unique-cuniquevalidator) that can check for duplicate values in the database, which is optimized for handling duplicate checks. Which also means the function `check_ducplicate()` will no longer needed. It can be implemented just like this in `protected/models/Project.php`:

```
public function rules()
{
    return array(
        array('url', 'required'),
        array('url', 'url', 'message' => 'Please check the URL format'),
        array('url', 'length', 'max' => 128),
        array('name', 'length', 'max' => 255),
        // Use the built-in unique validator:
        array('url', 'unique', 'message' => 'Duplicate URL'),
        array('name', 'unique', 'message' => 'Duplicate Project Name'),
    );
}
```
**suggestion:** Since this feature involves user interaction in the admin project page, I think acceptance test is more suitable for this.